internal/service/ipam/api_address_block_resource_test.go

```diff
@@ -221,6 +221,38 @@ func TestAccAddressBlockResource_Comment(t *testing.T) {
 	})
 }
 
+// The provider does not support creating user roles , which is a required field for creating a compartment
+// Hence, this test is completed with root compartment only
+func TestAccAddressBlockResource_CompartmentId(t *testing.T) {
+	var resourceName = "bloxone_ipam_address_block.test_compartment_id"
+	var v ipam.AddressBlock
+	spaceName := acctest.RandomNameWithPrefix("ip-space")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAddressBlockCompartmentId(spaceName, "192.168.0.0", "16", ""),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAddressBlockExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "compartment_id", ""),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccAddressBlockCompartmentId(spaceName, "192.168.0.0", "16", ""),
```

Comment:
since there is no change here, there will be no `update`

Reply:
can you also add a case for updating to `null` value after some compartment has been set. That should revert it to the root compartment (`""`) since the default is given.

Reply:
Done